src/services/request.service.js

```diff
@@ -215,10 +215,15 @@ const sendRequest = async (engine, requestUrl, method, authentication, proxy, da
 
   // Generate authentication header
   const headers = baseHeaders
-  if (authentication && (authentication.type === 'Basic')) {
-    const decryptedPassword = engine.encryptionService.decryptText(authentication.password)
-    const basic = Buffer.from(`${authentication.username}:${decryptedPassword}`).toString('base64')
-    headers.Authorization = `Basic ${basic}`
+  if (authentication) {
```

Comment:
should be a "switch" as we will likely have more type in the future.

Reply:
It also comes from another old review

Reply:
What I meant was to check once if authentication is present as it is now. What means JF I think is that inside this if, a switch(authentication.type) would be better.

Reply:
I'll do that, for the default case, should raise an error?